modules/workers/instanceconfig.tf

```diff
@@ -64,6 +64,23 @@ resource "oci_core_instance_configuration" "workers" {
         }
       }
 
+      dynamic "platform_config" {
+        for_each = each.value.platform_config != null ? [1] : []
+        content {
+          type                                           = lookup(each.value.platform_config, "type")
```

Comment:
Can we derive 'type' from 'shape' rather than exposing it as a variable? 
We can get type from data source oci_ocvp_supported_host_shapes https://registry.terraform.io/providers/oracle/oci/latest/docs/data-sources/ocvp_supported_host_shapes

Reply:
The ocvp data source is for OCVS, not for OKE. Does OKE API has something similar?

Reply:
No idea.

Reply:
It looks like [data-sources/core_shapes](https://registry.terraform.io/providers/oracle/oci/latest/docs/data-sources/core_shapes) has what we need:
```
            "name": "BM.GPU.B4.8",
...
            "platform_config_options": [
              {
                "access_control_service_options": [
                  {
                    "allowed_values": [
                      true,
                      false
                    ],
                    "is_default_enabled": false
                  }
                ],
                "input_output_memory_management_unit_options": [
                  {
                    "allowed_values": [
                      true,
                      false
                    ],
                    "is_default_enabled": false
                  }
                ],
                "measured_boot_options": [],
                "memory_encryption_options": [],
                "numa_nodes_per_socket_platform_options": [
                  {
                    "allowed_values": [
                      "NPS0",
                      "NPS1",
                      "NPS2",
                      "NPS4"
                    ],
                    "default_value": "NPS4"
                  }
                ],
                "percentage_of_cores_enabled_options": [],
                "secure_boot_options": [],
                "symmetric_multi_threading_options": [
                  {
                    "allowed_values": [
                      true,
                      false
                    ],
                    "is_default_enabled": true
                  }
                ],
                "trusted_platform_module_options": [],
                "type": "AMD_MILAN_BM_GPU",
                "virtual_instructions_options": [
                  {
                    "allowed_values": [
                      true,
                      false
                    ],
                    "is_default_enabled": false
                  }
                ]
              }
            ],
```

Reply:
Done.